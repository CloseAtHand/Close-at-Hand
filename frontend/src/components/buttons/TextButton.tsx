import { Pressable, StyleSheet, Text } from "react-native";
import { COLORS } from "../../shared/styles/STYLES";

interface Props {
	text: string;
	onPress: () => void;
}

const TextButton: React.FC<Props> = ({ text, onPress }) => {
	return (
		<Pressable
			style={({ pressed }) => [styles.container, pressed && styles.pressed]}
			onPress={onPress}
			testID="button"
		>
			<Text style={styles.text}>{text}</Text>
		</Pressable>
	);
};

export default TextButton;

const styles = StyleSheet.create({
	container: {
		justifyContent: "center",
		alignItems: "center",
		alignSelf: "center",
	},
	pressed: {
		opacity: 0.75,
	},
	text: {
		color: COLORS.PupleBlue,
		backgroundColor: COLORS.White,
	},
});
