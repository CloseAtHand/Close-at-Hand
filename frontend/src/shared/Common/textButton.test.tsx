import "@testing-library/jest-native/";
import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import TextButton from "./TextButton";

describe("TextButton", () => {
	const TEXT = "터치하세요";

	it("스냅샷 테스트", () => {
		const tree = renderer
			.create(<TextButton text={TEXT} onPress={jest.fn()} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("입력한 텍스트가 렌더링됩니다.", () => {
		const { getByText } = render(
			<TextButton text={TEXT} onPress={jest.fn()} />,
		);

		const text = getByText(TEXT);

		expect(text).toBeTruthy();
	});

	it("버튼을 누르면 onPress 함수가 실행됩니다.", () => {
		const onPress = jest.fn();

		const { getByTestId } = render(
			<TextButton text={TEXT} onPress={onPress} />,
		);

		const button = getByTestId("button");

		fireEvent.press(button);
		expect(onPress).toHaveBeenCalled();
	});
});
