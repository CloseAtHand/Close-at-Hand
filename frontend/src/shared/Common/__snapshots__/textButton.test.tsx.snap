// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton 스냅샷 테스트 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="button"
>
  <Text
    style={
      {
        "color": "#8DE0F8",
      }
    }
  >
    터치하세요
  </Text>
</View>
`;
